src/lib.rs

```diff
@@ -39,23 +42,49 @@ pub enum Error {
     RuntimeError(String),
 }
 
-pub struct KernelInterface {
-    run_command: Box<FnMut(&str, &[&str]) -> Result<Output, Error>>,
+pub trait CommandRunner {
+    fn run_command(&mut self, program: &str, args: &[&str]) -> Result<Output,Error>;
 }
 
-impl KernelInterface {
-    pub fn new() -> KernelInterface {
+pub struct MockCommandRunner<'a> {
+    counter: usize,
```

Comment:
This is very specific to the needs of the current tests, but I guess we can refactor as needed.